src/api.rs

```diff
@@ -86,8 +96,21 @@ pub async fn manifest(req: web::Path<ManifestRequest>, qstr: web::Query<Manifest
 	static HIT_COUNTER: Lazy<IntCounterVec> = Lazy::new(|| register_int_counter_vec!("manifest_cache_hits", "Number of manifests read from cache", &["namespace"]).unwrap());
 	static MISS_COUNTER: Lazy<IntCounterVec> = Lazy::new(|| register_int_counter_vec!("manifest_cache_misses", "Number of manifest requests that went to upstream", &["namespace"]).unwrap());
 
-	let max_age = config.upstream.lock().await.get(qstr.ns.as_deref())?.manifest_invalidation_time;
-	let namespace = qstr.ns.as_deref().unwrap_or_else(|| config.default_ns.as_ref());
+	let (namespace, image) = match qstr.ns.as_deref() {
+		Some(ns) => (ns, req.image.as_ref().trim_start_matches(&format!("{}/", ns))),
```

Comment:
Is trimming the namespace from the beginning required when the namespace is passed as an explicit parameter?  I think not trimming it would only cause a problem if someone configured containerd, which supports `?ns=`, but included namespace-prefixed paths - I would consider that a misconfiguration

Reply:
That's fair, I figured it didn't hurt to support both being used for some reason but I can clean that up, agree it would likely be a misconfiguration